independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/PublisherResponseHandler.java

```diff
@@ -200,11 +200,11 @@ public void handle(ResteasyReactiveRequestContext requestContext) throws Excepti
         }
     }
 
-    private void handleStreaming(ResteasyReactiveRequestContext requestContext, Multi<?> result, boolean json) {
-        result.subscribe().withSubscriber(new StreamingMultiSubscriber(requestContext, json));
+    private void handleStreaming(ResteasyReactiveRequestContext requestContext, Publisher<?> result, boolean json) {
+        result.subscribe(new StreamingMultiSubscriber(requestContext, json));
```

Comment:
Not very important, but in the case of `Multi` (and only in this case - which would duplicate the code), you can do:

```
result.subscribe().withSafeSubscriber(...)
```

In this case, Mutiny disables all the serialization / TCK checks.

Reply:
That's nice!

We can use that actually by checking if `result` is `Multi` and doing this special handling in such a case.

Reply:
Weird, I can't find `withSafeSubscriber` anywhere :)